src/uct/ib/ud/base/ud_ep.c

```diff
@@ -835,9 +835,21 @@ static void uct_ud_ep_rx_creq(uct_ud_iface_t *iface, uct_ud_neth_t *neth)
                        "creq->conn_sn=%d ep->conn_sn=%d",
                        ctl->conn_req.conn_sn, ep->conn_sn);
 
-    ucs_assertv_always(ctl->conn_req.path_index == ep->path_index,
-                       "creq->path_index=%d ep->path_index=%d",
-                       ctl->conn_req.path_index, ep->path_index);
+    if (uct_ib_iface_is_roce(&iface->super)) {
+        if (iface->super.config.roce_path_factor > 0) {
+            ucs_assertv_always(ctl->conn_req.path_index == ep->path_index,
+                               "creq->path_index=%d ep->path_index=%d",
+                               ctl->conn_req.path_index, ep->path_index);
+        }
+    } else {
+        ucs_assertv_always((ctl->conn_req.path_index %
```

Comment:
/* With IB, path_bits component must be same on local and remote ep*/

Reply:
done